core/service/pool.go

```diff
@@ -175,3 +177,22 @@ func (i *Instance) Proposal() market.ServiceProposal {
 func (i *Instance) State() State {
 	return i.state
 }
+
+type instanceSocket struct {
+	Id         string `json:"id"`
+	Status     string `json:"status"`
+	ProviderID string `json:"providerId"`
+	Type       string `json:"type"`
+}
+
+// SetState - Set instance state and send new state to websocket
+func (i *Instance) SetState(state State) {
+	i.state = state
+	proposal := i.Proposal()
+	websocket.Instance.ServiceUpdateStatusAction(instanceSocket{
```

Comment:
This needs to be reworked. Including a global variable here is definitely a non-solution. Hook up the websockets to service start/stop events via the EventBus and avoid adding dependencies where they are not needed. 

Can't remember if the events are already being published or not - either way - add the publishing part if required.